streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -620,6 +644,21 @@
                     "",
                     Importance.LOW,
                     APPLICATION_SERVER_DOC)
+            .define(ASSIGNMENT_BALANCE_FACTOR_CONFIG,
+                    Type.INT,
+                    1,
+                    Importance.LOW,
+                    ASSIGNMENT_BALANCE_FACTOR_DOC)
+            .define(ASSIGNMENT_MAX_EXTRA_REPLICAS_CONFIG,
+                    Type.INT,
+                    5,
+                    Importance.LOW,
+                ASSIGNMENT_MAX_EXTRA_REPLICAS_DOC)
+            .define(ASSIGNMENT_PROBING_REBALANCE_INTERVAL_MS_CONFIG,
+                    Type.LONG,
+                    10 * 60 * 1000L,
+                    Importance.LOW,
+                    ASSIGNMENT_PROBING_REBALANCE_INTERVAL_MS_DOC)
```

Comment:
req: Please add unit tests for the definitions.

Reply:
What do you mean by tests "for the definitions"?

Reply:
Have a look into `StreamsConfigTest`, e.g., `shouldSetDefaultBuiltInMetricsVersionIfNoneIsSpecified()`.

Reply:
I'm not sure I see the value in testing the default for every new config, isn't that just testing the code that generates/sets the defaults (`ConfigDef#define`)? We don't test eg `Consumer#subscribe` inside `StreamThreadTest`

Reply:
The value is that if anybody changes the default value or the allowed values by mistake, maybe during a refactoring, there is a test that shows the mistake. Of course it is not a 100% protection, but at least it won't go totally unnoticed.

Reply:
Alright, I'll add the tests. We're missing tests for every other config then 🙂